contrib/DEVELOPMENT_WORKFLOW.md

```diff
@@ -1,159 +1,328 @@
 
-#### Finishing a Release Branch
+- `Constants:` Constants are usually defined on a module level and written in all capital letters with underscores separating words.
 
-This involves releasing stable code and generating a new version for bittensor.
+- `Non-public Methods and Instance Variables:` Use a single leading underscore (_). This is a weak "internal use" indicator.
 
-1. Switch branch to master: `git checkout master`
-2. Merge release branch into master: `git merge --no-ff release/3.4.0/optional-descriptive-message`
-3. Tag changeset: `git tag -a v3.4.0 -m "Releasing v3.4.0: some comment about it"`
-4. Push changes to master: `git push origin master`
-5. Push tags to origin: `git push origin --tags`
+- `Strongly "private" methods and variables:` Use a double leading underscore (__). This triggers name mangling in Python.
 
-To keep the changes made in the __release__ branch, we need to merge those back into `staging`:
+### 3. Python Code
 
-- Switch branch to staging: `git checkout staging`.
-- Merging release branch into staging: `git merge --no-ff release/3.4.0/optional-descriptive-message`
+- `List Comprehensions:` Use list comprehensions for concise and readable creation of lists.
 
-This step may well lead to a merge conflict (probably even, since we have changed the version number). If so, fix it and commit.
+- `Generators:` Use generators when dealing with large amounts of data to save memory.
 
+- `Context Managers:` Use context managers (with statement) for resource management.
 
-#### Creating a hotfix branch
-1. Create branch from master: `git checkout -b hotfix/3.3.4/descriptive-message/creator's-name master`
-2. Update patch version: `./scripts/update_version.sh patch`
-3. Commit file changes with new version: `git commit -a -m "Updated version to 3.3.4"`
-4. Fix the bug and commit the fix: `git commit -m "Fixed critical production issue X"`
+- `String Formatting:` Use f-strings for formatting strings in Python 3.6 and above.
 
-#### Finishing a Hotfix Branch
+- `Error Handling:` Use exceptions for error handling whenever possible.
 
-Finishing a hotfix branch involves merging the bugfix into both `master` and `staging`.
 
-1. Switch branch to master: `git checkout master`
-2. Merge hotfix into master: `git merge --no-ff hotfix/3.3.4/optional-descriptive-message`
-3. Tag new version: `git tag -a v3.3.4 -m "Releasing v3.3.4: descriptive comment about the hotfix"`
-4. Push changes to master: `git push origin master`
-5. Push tags to origin: `git push origin --tags`
-6. Switch branch to staging: `git checkout staging`
-7. Merge hotfix into staging: `git merge --no-ff hotfix/3.3.4/descriptive-message/creator's-name`
-8. Push changes to origin/staging: `git push origin staging`
-9. Delete hotfix branch: `git branch -d hotfix/3.3.4/optional-descriptive-message`
+Notes for Source Coding
+-----------------------------
```

Comment:
This section can go and is superfluous, imo. Neurons will be removed in a subsequent release of bittensor and this will become outdated.